GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1960,6 +1960,14 @@ private void AddNotes()
 
         private void FindFileInSelectedCommit()
         {
+            var selectedRevisions = RevisionGrid.GetSelectedRevisions();
+            if (selectedRevisions.Count > 1 || (selectedRevisions.Count == 1 && selectedRevisions[0].IsArtificial))
+            {
+                GitRevision potentialRevision = selectedRevisions[0];
+                ObjectId targetCommit = potentialRevision.IsArtificial ? RevisionGrid.CurrentCheckout : potentialRevision.ObjectId;
```

Comment:
Will not work if artificial commit is in the multiselect
Not the most important use case though, but removing the artificial commits from selectedRevisions is not that hard...

Reply:
I will test again to be sure I didn't broke something when refactoring but I have tested this multi-commits with the 1st artificial and it was working...

Reply:
First artificial should work, but you could select two artificial commits

Reply:
For me this case is handled by the ternary in the if clause...

Reply:
I agree, Philippe. What have me missed, Gerhard?

The repetition can be avoided (and an empty selection be handled, too):
```
GitRevision? potentialRevision = selectedRevisions.FirstOrDefault();
ObjectId? targetCommit = potentialRevision?.IsArtificial is not false ? RevisionGrid.CurrentCheckout : selectedRevisions.Count == 1 ? null : potentialRevision!.ObjectId;
if (targetCommit is not null)
```

Reply:
Yes this works.
Just null annotation (is OK for SetSelectedRevision())

```suggestion
                ObjectId? targetCommit = potentialRevision.IsArtificial ? RevisionGrid.CurrentCheckout : potentialRevision.ObjectId;
```

Reply:
Sorry @mstv I find the LoC too difficult to understand...